chasm/lib/callback/invocation_tasks_test.go

```diff
@@ -0,0 +1,298 @@
+type invocationTasksSuite struct {
+	callbackSuite
+}
+
+func TestInvocationTasksSuite(t *testing.T) {
+	suite.Run(t, &invocationTasksSuite{})
+}
+
+func (s *invocationTasksSuite) SetupTest() {
+	s.callbackSuite.SetupTest()
+}
+
+// TestInvocationTask_SuccessfulNexusCallback tests a successful HTTP callback invocation.
+// Mirrors HSM test: TestProcessInvocationTaskNexus_Outcomes/success
+// TODO (seankane): This test is functionally a duplicate of TestExecuteInvocationTaskNexus_Outcomes/success
+// in executors_test.go until CHASM testing infrastructure matures to support verifying task generation
+// in integration tests. Once that's available this test will test the full CHASM stack integration.
+func (s *invocationTasksSuite) TestInvocationTask_SuccessfulNexusCallback() {
+	// Setup HTTP caller that returns 200 OK
+	s.httpCallerProvider = func(nid queues.NamespaceIDAndDestination) callback.HTTPCaller {
+		return func(r *http.Request) (*http.Response, error) {
+			return &http.Response{StatusCode: 200, Body: http.NoBody}, nil
+		}
+	}
+	s.setupDefaultDependencies()
+
+	// Create callback in SCHEDULED state
+	s.createCallback("cb-1", "req-1", "http://localhost", callbackspb.CALLBACK_STATUS_SCHEDULED)
+
+	// Create mutable context and get callback through it (like scheduler pattern)
+	ctx := s.newMutableContext()
+	callbackField := s.workflow.Callbacks["cb-1"]
+	cb, err := callbackField.Get(ctx)
+	s.NoError(err)
+
+	// Set up contexts for reading
+	readCtx := s.newContext()
+
+	// Set up engine expectations to use the real callback from the tree
+	s.ExpectReadComponent(readCtx, cb)
+	s.ExpectUpdateComponent(ctx, cb)
+
+	// Create engine context for side effect task execution
+	engineCtx := s.newEngineContext()
+
+	// Create ComponentRef
+	ref := chasm.NewComponentRef[*callback.Callback](chasm.EntityKey{
+		NamespaceID: s.tv.NamespaceID().String(),
+		BusinessID:  s.tv.WorkflowID(),
+		EntityID:    s.tv.RunID(),
+	})
+
+	// Execute InvocationTask with engine context
+	err = s.invocationExecutor.Execute(
+		engineCtx,
+		ref,
+		chasm.TaskAttributes{Destination: "http://localhost"},
+		&callbackspb.InvocationTask{Attempt: 0},
+	)
+	s.NoError(err)
+
+	// Close transaction to commit tasks to node backend
+	_, err = s.node.CloseTransaction()
+	s.NoError(err)
+
+	// Verify callback transitioned to SUCCEEDED
+	s.Equal(callbackspb.CALLBACK_STATUS_SUCCEEDED, cb.Status)
+}
+
+// TestInvocationTask_FailureTriggersBackoff tests that a retryable HTTP error triggers backoff.
+// Mirrors HSM test: TestProcessInvocationTaskNexus_Outcomes/retryable-error
+// TODO (seankane): This test is functionally a duplicate of TestExecuteInvocationTaskNexus_Outcomes/retryable-http-error
+// in executors_test.go until CHASM testing infrastructure matures to support verifying task generation
+// in integration tests. Once that's available this test will test the full CHASM stack integration.
+func (s *invocationTasksSuite) TestInvocationTask_FailureTriggersBackoff() {
```

Comment:
Why do you need to test this if we have `executors_test.go` and tests for the statemachine transition that where this task is added?